src/main/java/org/jenkins/plugins/lockableresources/actions/LockableResourcesRootAction.java

```diff
@@ -158,6 +160,36 @@ public int getAssignedResourceAmount(String label) {
     return LockableResourcesManager.get().getResourcesWithLabel(label, null).size();
   }
 
+  /** Returns current queue */
+  @Restricted(NoExternalUse.class) // used by jelly
+  public List<QueuedContextStruct> getCurrentQueuedContext() {
+    return LockableResourcesManager.get().getCurrentQueuedContext();
+  }
+
+  /** Returns current queue */
+  @Restricted(NoExternalUse.class) // used by jelly
+  @CheckForNull
+  public LockableResourcesStruct getOldestQueue() {
+    LockableResourcesStruct oldest = null;
+    for (QueuedContextStruct context : this.getCurrentQueuedContext()) {
+      for(LockableResourcesStruct resourceStruct : context.getResources()) {
+        if (resourceStruct.ticks == 0) {
+          // Older versions of this plugin might miss this information.
+          // Therefore skip it here.
+          continue;
+        }
+        if (oldest == null) {
+          oldest = resourceStruct;
+          continue;
+        }
+        if (oldest.ticks > resourceStruct.ticks) {
+          oldest = resourceStruct;
+        }
```

Comment:
```suggestion
        if (oldest == null || oldest.ticks > resourceStruct.ticks) {
          oldest = resourceStruct;
        }
```

Reply:
Can you use the `LockableResourcesStruct::getQueuedTimestamp()` method here? In this case, you could make the `ticks` variable (or renamed variable, see comment below) private.